java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -301,35 +355,52 @@ public static boolean isInitialized() throws RmmException {
    * allocator decides to return more memory than what was requested. However,
    * the result will always be a lower bound on the amount allocated.
    */
-  public static synchronized long getTotalBytesAllocated() {
-    if (tracker == null) {
-      return 0;
-    } else {
-      return tracker.getTotalBytesAllocated();
+  public static long getTotalBytesAllocated() {
+    readLock.lock();
+    try {
+      if (tracker == null) {
+        return 0;
+      } else {
+        return tracker.getTotalBytesAllocated();
+      }
+    } finally {
+      readLock.unlock();
     }
   }
 
+
   /**
    * Returns the maximum amount of RMM memory (Bytes) outstanding during the
    * lifetime of the process.
    */
-  public static synchronized long getMaximumTotalBytesAllocated() {
-    if (tracker == null) {
-      return 0;
-    } else {
-      return tracker.getMaxTotalBytesAllocated();
+  public static long getMaximumTotalBytesAllocated() {
+    readLock.lock();
+    try {
+      if (tracker == null) {
+        return 0;
+      } else {
+        return tracker.getMaxTotalBytesAllocated();
+      }
+    } finally {
+      readLock.unlock();
     }
   }
 
+
   /**
    * Resets a scoped maximum counter of RMM memory used to keep track of usage between
    * code sections while debugging.
    *
    * @param initialValue an initial value (in Bytes) to use for this scoped counter
    */
-  public static synchronized void resetScopedMaximumBytesAllocated(long initialValue) {
-    if (tracker != null) {
-      tracker.resetScopedMaxTotalBytesAllocated(initialValue);
+  public static void resetScopedMaximumBytesAllocated(long initialValue) {
+    writeLock.lock();
```

Comment:
I think this can be a read lock

Reply:
agree there is a lock already in the impl here.